### ISSUE TITLE
Problematic normalization

### ISSUE THREAD

**Issue by Coco-hanqi** [matyasbohacek/spoter#3]
![Screen Shot 2022-02-13 at 5 36 52 PM](https://user-images.githubusercontent.com/92771366/153785939-419be1b8-87d1-4f37-8232-3b9d74873505.png)
Got a validation accuracy around 58%, lower than the one proposed in the paper. Is the lower accuracy caused by this problematic normalization error?

**Comment by matyasbohacek** [comment]
Thanks for opening this ticket, @Coco-hanqi.

Yes, this seems strange, probably related to the under-performing accuracy. Let me try to reproduce this behavior on a new machine and investigate the source of the problem -- will get back to you here shortly.

**Comment by caicairay** [comment]
I also got this problem. It seems because of the problematic key points data. The left and right shoulder key points are overlapped in some frames.

**Comment by matyasbohacek** [comment]
Dear @Coco-hanqi & @caicairay,

I finally managed to reproduce this on an older external machine. Upon further research, it seems that some PyTorch versions seem to have various low-level issues with model checkpointing via saving the state dictionary. Interestingly enough, the same code with no edits worked on my macOS machine. Hence, the script has been rewritten to keep the entire model objects, which managed to solve this on the Linux instance where I reproduced this.

Please, let me know if this issue persists, and we can look into this deeper. Feel free to reach out should any other problems or questions arise.

**Comment by andresherrera97** [comment]
I'm getting the same issue of the Problematic Normalization on a Ubuntu 22 machine, any update on this?

**Comment by andresherrera97** [comment]
I realized that Problematic Normalization occurs during training when the augmentation happens to be rotation. Disabling the option of rotation prevents the Problematic Normalization and accuracy on test increases to 60%, which is still below the results in the paper

**Comment by saurus2** [comment]
Hi, 
I had a same problem.
The Validation accuracy was not over 60%.

I used MacOS though I got a problem.
Then I used my campus HPC server.
Then I got same problems. 

I used WLASL 100 25fps dataset.
Somebody have an idea?

**Comment by ShortCatislong** [comment]
> 



> I realized that Problematic Normalization occurs during training when the augmentation happens to be rotation. Disabling the option of rotation prevents the Problematic Normalization and accuracy on test increases to 60%, which is still below the results in the paper

could you please kindly remind me which rotation method you have disabled?
the __rotate() , augment_rotate() or augment_arm_joint_rotate()?
